File: projects/10/ruby/Owl.rb
# Owl.rb | Defines the Owl class which inherits attributes and methods
#   from the Bird superclass.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Sarah Whitten
# Date: April 20, 2019
####################################################

require './Bird.rb'
require './FlyingBird'

class Owl < FlyingBird

	def call
    	'Hoo!'
	end

end

File: projects/10/java/Owl.java
/* Owl.java provides an Owl subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Sarah Whitten
 * Date: April 20, 2019
 ******************************************************/

public class Owl extends FlyingBird
{
	/* default constructor */
	public Owl()
	{
		super();
	}

	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */
	public Owl(String name)
	{
		super(name);
	}

	/* An Owl's Call
	 * Return: an owl-call ("Whoo-hoo!").
	 */
	public String call()
	{
		return "Hoo!";
	}
} 

